src/DataCollection.Shared.Tests/DataCollection.Shared.Tests.csproj

```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DefineConstants>TRACE;DOT_NET_CORE_TEST</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>TRACE;DOT_NET_CORE_TEST</DefineConstants>
+  </PropertyGroup>
```

Comment:
Since this is set on all configs, perhaps just add the DefineConstants under line 4? Also good practice is to only add rather than replace, so you won't overwrite `DEBUG`. ie  `<DefineConstants>$(DefineConstants);TRACE;DOT_NET_CORE_TEST</DefineConstants>`